### PR TITLE
Effy small UI polish

### PR DESCRIPTION
**Pull request by effyyzhang**
Adds a new onboarding screen which gates a user from visiting the application's home page until they've added an ai api key:

Screenshot 2024-10-24 at 3 59 06 PM

### CODE REVIEW

File: packages/components/tsconfig.json [hunk 1]
```diff
@@ -7,6 +7,6 @@
     "jsx": "react-jsx",
     "baseUrl": "."
   },
-  "include": ["src/index.tsx", "src/**/*.ts", "src/**/*.tsx"],
+  "include": ["src/index.tsx", "src/**/*.ts", "src/**/*.tsx", "../web/src/components/onboarding-modal.tsx"],
```
Comment (benjreinhart) [inline packages/components/tsconfig.json:10]: Why this change? FWIW we cannot have files in the `components` package reference files outside it since it does not depend on web (and can't since that's circular dep). Also FWIW, we're going to remove the components package as that was introduced for reasons that no longer exist.

File: packages/web/src/components/apps/markdown.tsx [hunk 2]
```diff
@@ -1,10 +1,11 @@
...
+import { cn } from '@/lib/utils';
 
 export default function Markdown(props: { source: string; className?: string }) {
   return (
-    <div className={cn('sb-prose-chat px-2', props.className)}>
-      <MarkdownReact gfm>{props.source}</MarkdownReact>
-    </div>
+    <div
+      className={cn('sb-prose-chat px-2', props.className)}
+      dangerouslySetInnerHTML={{ __html: marked(props.source) }}
```
Comment (benjreinhart) [inline packages/web/src/components/apps/markdown.tsx:8]: Why this change? Also, we need gfm.

File: packages/web/src/routes/home.tsx [hunk 3]
```diff
@@ -66,6 +68,29 @@ export default function Home() {
   const [appToDelete, setAppToDelete] = useState<AppType | null>(null);
   const [showCreateAppModal, setShowCreateAppModal] = useState(false);
 
+  const { aiProvider, openaiKey, anthropicKey } = useSettings();
+  const [showOnboarding, setShowOnboarding] = useState(false);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    const checkApiKey = () => {
+      const hasSeenOnboarding = localStorage.getItem('hasSeenOnboarding') === 'true';
```
Comment (benjreinhart) [inline packages/web/src/routes/home.tsx:77]: This should live in persistent storage (sqlite), at least in the medium-long term.

**Review by benjreinhart — COMMENTED** [review]
Can we use css for the gradient? The image seems unnecessary and adds to the package sizes.

File: packages/web/src/routes/home.tsx (hunk 3 above)
Comment (1egoman) [reply packages/web/src/routes/home.tsx]: I updated this logic to instead gate on whether `aiEnabled` being outputted from `useSettings` was enabled rather than segregating the onboarding state individually.

I don't think there's a reason that tracking this separately is important, given that this should be shown for all users that don't have an ai api key set - if so though, please call it out.

**Review by benjreinhart — APPROVED** [review]
